oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java

```diff
@@ -97,19 +106,38 @@ public void in(MetaInAlarm meta, Metrics metrics) {
             return;
         }
 
+        final String metaName = meta.getName();
         if (CollectionUtils.isNotEmpty(includeNames)) {
-            if (!includeNames.contains(meta.getName())) {
+            if (!includeNames.contains(metaName)) {
                 if (log.isTraceEnabled()) {
-                    log.trace("{} isn't in the including list {}", meta.getName(), includeNames);
+                    log.trace("{} isn't in the including list {}", metaName, includeNames);
                 }
                 return;
             }
         }
 
         if (CollectionUtils.isNotEmpty(excludeNames)) {
-            if (excludeNames.contains(meta.getName())) {
+            if (excludeNames.contains(metaName)) {
+                if (log.isTraceEnabled()) {
+                    log.trace("{} is in the excluding list {}", metaName, excludeNames);
+                }
+                return;
+            }
+        }
+
+        if (includeNamesRegex != null) {
+            if (!includeNamesRegex.matcher(metaName).matches()) {
+                if (log.isTraceEnabled()) {
+                    log.trace("{} doesn't match the include regex {}", metaName, includeNamesRegex);
+                }
+                return;
+            }
+        }
+
+        if (excludeNamesRegex != null) {
```

Comment:
Just a reminder, as we are using JDK8+, Optional could replace all these `!=null`. But both of these styles exist today. So, don't require change, anyway.

Reply:
Thanks, I will change to this way since next PR.